### DISCUSSION TITLE
Specific formats for requests that embed SQL?

### DISCUSSION THREAD

**Discussion by KLABDB3K** [PygmalionOfCyprus/cmo-db-requests#1262] [General]
I am probably showing my ignorance of all things SQL but I have noticed Claudedev is using a very specific format for requests. 
Are these embedding some form of SQL link to enable the request to be inputted more quickly?
If so how do you find that formatting protocol or links to how to learn it please?
Regards
K

**Comment by PygmalionOfCyprus** [comment]
@claudejdev

**Comment by claudejdev** [comment]
Hey there K,

thanks to Hethan's answers, I've learned that an SQL command is useful to the DB gang when there is a repetition of edits to be done on a handful of platforms.
A recent example was to modify the weight and range, to the same value, on a bunch of AMRAAMs.
Note that it work on a given _table_, no on multiple tables at once.
(Now, to figure out what's what in the DB, I simply browse it using a simple free tool such as  [DB Browser for SQLite](https://sqlitebrowser.org/) )
Then I formulate my SQL statement in between _code_ tags, ```` aka "backticks". Just for readability, really. It would work all the same set in Bold, Italics, or even without formatting at all.

eg.
```
UPDATE DataWeapon SET DataWeapon.Weight=157.8,DataWeapon.AirRangeMax=27.0,DataWeapon.AirRangeMin=1.1
WHERE DataWeapon.ID IN (53,417,446,946,1383);
```
So I want to UPDATE the table named DataWeapon. Because I've looked it up and know that's where all the data I'd suggest a change to are living.
![image](https://user-images.githubusercontent.com/35732829/171512199-f805773c-5bcc-453d-849e-db8a86c78a4f.png)
![Capture d’écran (657)](https://user-images.githubusercontent.com/35732829/171512639-1a631415-4490-497b-ac62-6d789f815b5d.png)

In which I would like to SET the column Weight, of the DataWeapon table, thus "DataWeapon.Weight, to a value equal to " 157.8".
I just write the other columns and values similarly, separated by a comma, but at the end of the line.

Now, we have to tell for which platforms should these changes be applied.
For that, I like to use the IN operator and then just list, between parenthesis, the platform's ID separated by a comma. As simple as it can get.
Oh, and don't never ever forget the closing semicolon. It might not be mandatory in all DB systems, but I've heard that a random kitten is smashed every time we do forget to close our SQL statement. So, yeah, let's just not.


Note it's no silver bullet, and actually useful in a very specific use case. But in that niche, an SQL statement is king of the pond, and helps the DB Dream Team to handle it quicker than modifying each platform with the same modification, as far as I know.

If you want to practice, and understand, SQL, I couldn't recommend enough _[W3Schools Online Web Tutorials](https://www.w3schools.com/)_. There are many like it, but this one I ended up learning a lot from. It might even be my best friend on my journey to master SQL.
I would, specifically, suggest the [SQL UPDATE Statement](https://www.w3schools.com/sql/sql_update.asp) page.

For ref, here are Ethan's diligent answers to my cumbersome questions :

https://github.com/PygmalionOfCyprus/cmo-db-requests/issues/918#issuecomment-1101400254
https://github.com/PygmalionOfCyprus/cmo-db-requests/issues/986#issuecomment-1109391140
https://github.com/PygmalionOfCyprus/cmo-db-requests/issues/1252#issuecomment-1140770225

PS: had a long day, so my answer might be muddy. I'll refine it asap. Meanwhile, feel free to reach out to me on Discord or on Matrix's forum. I'll gladly help you with whatever I know at this point.

**Reply by PygmalionOfCyprus** [reply]
I'll also add that, if you're not entirely sure of your SQL or just want to add a backup, it's almost as helpful just to give a formatted list of IDs (ideally separated by commas) along with whatever you want changed.

So, for example, what Claude did in #918 is perfectly fine.

**Reply by KLABDB3K** [reply]
Excellent, many thanks, that's going to take some reading to absorb, but brilliant and exactly what I was asking for. 
Thank you for such a detailed answer.
